File: EduChemSuite.API/EduChemSuite.API/Models/QuestionModel.cs
namespace EduChemSuite.API.Models;

public class QuestionModel : BaseModel
{
    public required Guid UserId { get; set; }
    public UserModel? User { get; set; }
    public required String QuestionText { get; set; }
    public required Guid QuestionTypeId { get; set; }
    public virtual QuestionTypeModel? QuestionType { get; set; }
    public virtual ICollection<QuestionTagModel>? QuestionTags { get; set; }
}

File: EduChemSuite.UI/UI/src/app/_models/QuestionModel.ts
import {BaseModel} from "./BaseModel";
import {UserModel} from "./UserModel";
import {QuestionTypeModel} from "./QuestionTypeModel";
import {QuestionTagModel} from "./QuestionTagModel";
import {AnswerModel} from "./AnswerModel";

export class QuestionModel extends BaseModel {
  userId?: string;
  user?: UserModel;
  questionText?: string;
  questionTypeId?: string;
  questionType?: QuestionTypeModel;
  questionTags?: QuestionTagModel[];
  answers?: AnswerModel[];
  version?: number;
}
